biz.aQute.resolve/src/biz/aQute/resolve/ResolveProcess.java

```diff
@@ -209,6 +210,27 @@ public static ResolutionException augment(ResolveContext context, ResolutionExce
 		return augment(re.getUnresolvedRequirements(), context, re);
 	}
 
+	public static String format(Collection<Requirement> requirements) {
+		try (Formatter f = new Formatter()) {
+			for (Requirement req : requirements) {
+				boolean optional = isOptional(req);
+				f.format("%n %s[%-19s] %s%s", isOptional(req) ? " " : "*", req.getNamespace(), req,
```

Comment:
`*` again? How is anyone supposed to know what the `*` means? I thought you had planned to partition the output such that all mandatory requirements come before all optional requirements. To me, that would be more useful than `*` since it clearly groups things.